atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/keyvalue/impl/AbstractKeyValueServiceTest.java

```diff
@@ -972,6 +972,64 @@ public void shouldAllowRemovingAllCellsInDynamicColumns() {
         assertThat(values, is(emptyMap()));
     }
 
+
+    @Test
+    public void shouldAllowSameTablenameDifferentNamespace() {
+        TableReference fooBar = TableReference.createUnsafe("foo.bar");
+        TableReference bazBar = TableReference.createUnsafe("baz.bar");
+
+        // try create table in same call
+        keyValueService.createTables(
+                ImmutableMap.of(
+                        fooBar, AtlasDbConstants.GENERIC_TABLE_METADATA,
+                        bazBar, AtlasDbConstants.GENERIC_TABLE_METADATA));
+
+        // try create table spanned over different calls
+        keyValueService.createTable(fooBar, AtlasDbConstants.GENERIC_TABLE_METADATA);
+        keyValueService.createTable(bazBar, AtlasDbConstants.GENERIC_TABLE_METADATA);
+
+        // test table actually created
+        keyValueService.getRange(fooBar, RangeRequest.all(), 0L);
```

Comment:
well, you made me learn hamcrest at least.
it needs to be "containsInAnyOrder" because hamcrest version of matcher.contains() is a little troll-y.